pkg/apis/serving/v1beta1/trained_model.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2020 kubeflow.org.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// TrainedModel is the Schema for the TrainedModel API
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="URL",type="string",JSONPath=".status.url"
+// +kubebuilder:printcolumn:name="Ready",type="string",JSONPath=".status.conditions[?(@.type=='Ready')].status"
+// +kubebuilder:printcolumn:name="Age",type="date",JSONPath=".metadata.creationTimestamp"
+// +kubebuilder:resource:path=trainedmodel,shortName=tm
+type TrainedModel struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              TrainedModelSpec   `json:"spec,omitempty"`
+	Status            TrainedModelStatus `json:"status,omitempty"`
+}
+
+// TrainedModelSpec defines the trained model spec
+type TrainedModelSpec struct {
+	// parent inference service to deploy to
+	// +required
+	InferenceService string `json:"inferenceService"`
+	// Predictor model spec
+	// +required
+	PredictorModel ModelSpec `json:"predictorModel"`
+}
+
+// ModelSpec describes a trained model
+type ModelSpec struct {
+	// Storage URI for the model repository
+	StorageURI string `json:"storageUri"`
+	// Machine Learning framework which is used for trained model
+	// Valid values are:
```

Comment:
Does this enum map to our Predictor implementations? Can we codify this in code somehow?

Reply:
We will have trained model validation webhook which codifies this, see details [here](https://github.com/kubeflow/kfserving/issues/1006)